lib/logstash/outputs/bson/logstash_event.rb

```diff
@@ -0,0 +1,76 @@
+# Copyright (C) 2009-2014 MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Modified 2015 Elastic
+
+module BSON
+
+  # Injects behaviour for encoding and decoding time values to
+  # and from raw bytes as specified by the BSON spec.
+  #
+  # @see http://bsonspec.org/#/specification
+  module LogStashEvent
+
+    # An Event is an embedded document is type 0x03 in the BSON spec..
+    BSON_TYPE = 3.chr.force_encoding(BINARY).freeze
+
+    # Get the event as encoded BSON.
+    # @example Get the hash as encoded BSON.
+    #   Event.new("field" => "value").to_bson
+    # @return [ String ] The encoded string.
+    # @see http://bsonspec.org/#/specification
+     def to_bson(buffer = ByteBuffer.new)
+      position = buffer.length
+      buffer.put_int32(0)
+      to_hash_with_metadata.each do |field, value|
+        buffer.put_byte(value.bson_type)
```

Comment:
What do you think about adding some logging around here about this topic? clear feedback would be nice to me.